File: trunk/src/tools/genericappsimulation/php/iome/iogenericsimulation.php
<?php

class iogenericsimulation {
 
        var $name;
	private $params = array();
	private $metadata = array();
 
	public function __construct() {
	}
 
	public function addmetadata($prop, $name) {
                $metad=new iometadata($prop, $name)
		array_push($this->params, $metad);
	}

	public function setmetadata($prop, $name) {
		foreach ($metadata as $value)
		  {
		  if( ($value->name) == $name) $value->property=$prop;
		  } 
	} 

	public function getmetadata($name) {
                foreach ($metadata as $value)
		  {
		  if( ($value->name) == $name) return $value->property;
		  } 
  
		return;
	}        

	public function readsimulation($filename) {

		$parser=new iosimreader();
                $status=$parser->readsimulation($filename, $this);
		return $status;
	}

	public function writesimulation($filename) {

		$writer=new iosimwriter($this);
                $status=$writer->writesimulation($filename);
		return $status;
	}
 
}




?>

File: trunk/src/tools/genericappsimulation/python/iogenericsimulation.py
#!/usr/bin/python


#Run this program 
import os
import sys
import xml.sax
from iometadata import *
from iosimreader import *
from iosimwriter import *


# genericsimulation
class iogenericsimulation:
    name=""
    metadata=[]
    params=[]
    def __init__(self, iname):
        self.name=iname
    def addmetadata(self, prop,name):
        metad=iometadata(prop,name)
        self.metadata.append(metad)
    def addprop(self, prop):
        self.params.append(prop)
    def readsimulation(self,filename):
        parser = xml.sax.make_parser()
        handler = iosimreader.simhandler()
        parser.setContentHandler(handler)
        parser.parse(filename)
    def writesimulation(self,filename):
        writer=iosimwriter(self)
        status=writer.writesimulation(filename)
	return status

